extension/app/src/components/conversation/UserMessage.tsx

```diff
@@ -50,6 +77,86 @@ export function UserMessage({
     []
   );
 
+  const configurationIds = message.mentions.map(
+    (mention) => mention.configurationId
+  );
+
+  useEffect(() => {
+    const loadConfigurations = async () => {
+      const saved = await getSavedConfigurations();
+      if (saved.length !== savedConfigurations.length) {
+        setSavedConfigurations(saved);
+      }
+    };
+    void loadConfigurations();
+  }, []);
+
+  useEffect(() => {
+    const getId = async () => {
+      const hash = await hashBase64(`${configurationIds}-${message.content}`);
+      const id = `ask_dust_${hash}`;
+      setSavedConfigurationId(id);
+    };
+    void getId();
+  }, []);
+
+  const enabled = !!savedConfigurations.find(
+    (c) => c.id === savedConfigurationId
+  );
+
+  const buttons = isFirstMessage
+    ? [
+        <Button
+          variant="outline"
+          size="xs"
+          icon={enabled ? HeartIcon : HeartStrokeIcon}
+          label={enabled ? "Remove from context menu" : "Save to context menu"}
+          onClick={async () => {
+            if (enabled) {
+              const configurations = savedConfigurations.filter(
+                (c) => c.id !== savedConfigurationId
+              );
+              void sendMessage({
+                type: "UPDATE_SAVED_CONFIGURATIONS",
+                configurations,
+              });
+              setSavedConfigurations(configurations);
+            } else {
+              if (savedConfigurationId) {
+                const text = elRef.current?.innerText || "";
+                const description =
+                  text.length > 30 ? text.substring(0, 30) + "..." : text;
```

Comment:
NIT: could we do more than 30 chars? 🙏🏻 
Also, there's a special char for ellipsis, you can get it from https://www.compart.com/en/unicode/U+2026

Reply:
Actually we don't need to add an ellipsis, chrome handles it by utself (smthg around 70 chars)